mqtt/mqtt-broker/src/broker.rs

```diff
@@ -100,6 +104,52 @@ where
         Ok(self.snapshot())
     }
 
+    fn reevaluate_subscriptions(&mut self) -> Result<(), Error> {
+        let mut client_ids_to_remove: Vec<ClientId> = Vec::new();
+        for (client_id, session) in &self.sessions {
+            for sub in session
+                .subscriptions()
+                .into_iter()
+                .flat_map(std::collections::HashMap::values)
+            {
+                let client_info = session.client_info()?.clone();
```

Comment:
this can error out for offline sessions, but at the same time we need to have client_info to perform authorization... @dmolokanov, do you think we can save "last used" client info in offline session?

Reply:
Yeah, it was not the requirement previously, we probably need this info stored

Reply:
I've dealt with this for now by filtering out the offline case. Easiest way was to make a new method in Session. We can remove this when we do the refactor to store ths last used client info in offline session